File: Src/java/engine-jackson/src/test/java/org/opencds/cqf/cql/engine/execution/SortDescendingTest.java
package org.opencds.cqf.cql.engine.execution;

import java.util.List;

import org.opencds.cqf.cql.engine.elm.execution.EquivalentEvaluator;
import org.testng.Assert;
import org.testng.annotations.Test;

public class SortDescendingTest extends CqlExecutionTestBase {
    @Test
    public void testInterval() {
        Context context = new Context(library);

        Object result = context.resolveExpressionRef("sorted list of numbers descending").getExpression().evaluate(context);
        Assert.assertTrue(EquivalentEvaluator.equivalent(((List<?>) result).get(0), 9));
        Assert.assertTrue(EquivalentEvaluator.equivalent(((List<?>) result).get(1), 4));
        Assert.assertTrue(EquivalentEvaluator.equivalent(((List<?>) result).get(2), 2));
        Assert.assertTrue(EquivalentEvaluator.equivalent(((List<?>) result).get(3), 1));
    }
}

File: Src/java/engine/src/jvmTest/kotlin/org/opencds/cqf/cql/engine/execution/SortDescendingTest.kt
package org.opencds.cqf.cql.engine.execution

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.opencds.cqf.cql.engine.elm.executing.EquivalentEvaluator
import org.opencds.cqf.cql.engine.runtime.List
import org.opencds.cqf.cql.engine.runtime.toCqlInteger

internal class SortDescendingTest : CqlTestBase() {
    @Test
    fun evaluate() {
        val results = engine.evaluate { library("SortDescendingTest") }.onlyResultOrThrow
        val value = results["sorted list of numbers descending"]!!.value
        Assertions.assertTrue(
            EquivalentEvaluator.equivalent((value as List).elementAt(0), 9.toCqlInteger()).value ==
                true
        )
        Assertions.assertTrue(
            EquivalentEvaluator.equivalent(value.elementAt(1), 4.toCqlInteger()).value == true
        )
        Assertions.assertTrue(
            EquivalentEvaluator.equivalent(value.elementAt(2), 2.toCqlInteger()).value == true
        )
        Assertions.assertTrue(
            EquivalentEvaluator.equivalent(value.elementAt(3), 1.toCqlInteger()).value == true
        )
    }
}
